code/core/src/common/js-package-manager/PNPMProxy.test.ts

```diff
@@ -154,10 +154,10 @@ describe('PNPM Proxy', () => {
       expect(version).toEqual('5.3.19');
     });
 
-    it('throws an error if command output is not a valid JSON', async () => {
+    it('with constraint it throws an error if command output is not a valid JSON', async () => {
       vi.spyOn(pnpmProxy, 'executeCommand').mockResolvedValueOnce('NOT A JSON');
 
-      await expect(pnpmProxy.latestVersion('@storybook/core')).rejects.toThrow();
+      await expect(pnpmProxy.latestVersion('@storybook/core', '5.X')).rejects.toThrow();
     });
```

Comment:
The behavior without constraint cannot be tested because it does not throw an error nor does it attempt to validate the correctness of the command output.